src/registers/debug.rs

```diff
@@ -0,0 +1,496 @@
+//! Functions to read and write debug registers.
+
+#[cfg(feature = "instructions")]
+use core::arch::asm;
+use core::ops::Range;
+
+use bit_field::BitField;
+use bitflags::bitflags;
+
+/// Debug Address Register
+///
+/// Holds the address of a hardware breakpoint.
+pub trait DebugAddressRegister {
```

Comment:
Could we add an associated constant to this trait, something like:

```rust
const NUM: DebugAddressRegisterNumber;
```

Reply:
I added `DebugAddressRegister::NUM` by making `read` and `write` of the trait depend on the `instructions` feature. Is that the way to go, or should I rather make a new trait for `NUM`?